test/util/listener/listener.go

```diff
@@ -36,10 +38,18 @@ func (l *Listener) Accept() (net.Conn, error) {
 }
 
 func (l *Listener) Close() error {
-	if !l.closed {
+	l.RLock()
+	isListenerClosed := l.closed
+	l.RUnlock()
+
+	if !isListenerClosed {
 		close(l.c)
+
+		l.Lock()
 		l.closed = true
+		l.Unlock()
 	}
+
```

Comment:
The name should probably not be a question but a statement, like `listenerIsClosed`, it reads better in the if.

Reply:
Agree. Removed the variable as I don't need it when locking the whole method. Thanks!